geomstats/geometry/symmetric_matrices.py

```diff
@@ -228,9 +228,13 @@ def apply_func_to_eigvals(mat, function, check_positive=False):
         """
         eigvals, eigvecs = gs.linalg.eigh(mat)
         if check_positive and gs.any(gs.cast(eigvals, gs.float32) < 0.0):
-            logging.warning(
-                "Negative eigenvalue encountered in" " {}".format(function.__name__)
-            )
+            try:
```

Comment:
what is this? and does this have to be coded with a try/except? why not if and raise Error?

Reply:
I got an error from this line when function is a list of functions (in powerm).